pkg/controller/grafanadashboard/dashboard_controller.go

```diff
@@ -132,6 +133,12 @@ func (r *ReconcileGrafanaDashboard) Reconcile(request reconcile.Request) (reconc
 		return r.deleteDashboard(cr)
 	}
 
+	jsonTest := isJSON(cr.Spec.Json)
+	if jsonTest == false {
+		log.Info(fmt.Sprintf("found dashboard '%s/%s' but JSON is not valid", cr.Namespace, cr.Name))
```

Comment:
The owner of the dashboard does probably not have access to the operator logs, so they would not be able to figure out why their dashboard is not imported. Dashboards support status messages: https://github.com/integr8ly/grafana-operator/blob/master/pkg/apis/integreatly/v1alpha1/grafanadashboard_types.go#L21

It might be worth adding that log message to the dashboard as a status message. Also, is there any chance of getting a more detailed error from the parser, e.g. in which line it failed?

Reply:
I've added the JSON error to the status of the CR. I can't find a way to print out the line but this will print the error, which might be good enough?

Reply:
Yes, i think that's good enough. At least they know that the problem is with the json.